src/card/DebitCard.java

```diff
@@ -255,7 +308,22 @@ public boolean addCard(ConnectionSource databaseConnection) throws Exception {
 				throw(new Exception("Address not added properly, abort add card"));
 			}
 		}
-
+		
+		// validate inputs
+		if (!validateCardNumber(this.getCardNumber())) {
+			throw(new Exception("Card number not in the correct format"));
```

Comment:
these error messages should say what the correct format is. Ie a user might be putting in xxxx-xxxx-xxxx-xxxx in for their credit card number not knowing that they shouldn't be putting in dashes